db/sqldb/blob.go

```diff
@@ -233,11 +233,9 @@ func validateBlobForCreation(b *pacta.Blob) error {
 	if b.BlobURI == "" {
 		return fmt.Errorf("blob missing BlobURI")
 	}
-	if b.FileType == "" {
-		return fmt.Errorf("blob missing FileType")
-	}
 	if b.FileName == "" {
 		return fmt.Errorf("blob missing FileName")
 	}
+	// A blank FileType is valid, just means we don't recognize the file type yet.
```

Comment:
I'd disagree with that - we require a file type to be set, and I think that's a pretty straightforward invariant to maintain for the course of development. Is there a reason to support this?

Reply:
The context here is that the `create-report` action generates a whole directory of files, including HTML, JSON, CSS, JavaScript, fonts, and text files. I imagine the PACTA folks will want to modify these reports over time, potentially introducing new files and resources that should be served as part of the report.

If one of those new files is of a type we haven't enumerated (videos? various image formats? XML files? etc), we still want to persist the file and make a best-effort attempt to serve it. My solution is to say "well, we didn't recognize the type of the file at the time of report creation", which is `UNKNOWN`.

Happy to chat more if that doesn't make sense or you have another preferred approach.

Reply:
Yeah I guess I'm not sold there. In my mind the value of an enum is confidence you'll get back a value you expect. I think if we (at root) don't trust the data we have coming in to be of the type(s) that we expect, and we don't have a need to handle them specifically, we should simply remove the enum constraint generally, and store it as a string-based file extension.

Reply:
I don't want to throw the baby out with the bathwater here. In most cases (currently all), the blob is one of a small list of types. It's just that the alternative when we encounter an unknown extension is to ignore the file entirely, which is definitely not the option we want.

And the file type enum is more generally useful than a string extension. For example, `yml` and `yaml` are both YAML files